Dockerfile.dapper

```diff
@@ -15,6 +14,7 @@ ENV DAPPER_SOURCE /go/src/github.com/rancher/webhook/
 ENV DAPPER_OUTPUT ./bin ./dist
 ENV DAPPER_DOCKER_SOCKET true
 ENV DAPPER_RUN_ARGS "-v webhook-pkg:/go/pkg -v webhook-cache:/root/.cache"
+ENV GOPATH /go
```

Comment:
Would you mind explaining this line a bit? From the docs, the GOPATH should default to `$HOME/go` for the user that this is running under - why do we want to specify a custom location?

Reply:
You can omit this one, the next version of the go image will have this setting included.

Reply:
This was a tricky one to find the issue and fix it. We set `ENV DAPPER_SOURCE /go/src/github.com/rancher/webhook/` and the previous image `golang:1.17` set by default `GOPATH=/go`.

```bash
% docker run --rm -it golang:1.17 env
PATH=/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
GOLANG_VERSION=1.17.9
GOPATH=/go
TERM=xterm
HOME=/root
```

However `bci/golang` doesn't set `GOPATH`, so the packages will end up in `$HOME/go` while the code being compiled is in `ENV DAPPER_SOURCE /go/src/github.com/rancher/webhook/`. This was breaking the build.

```bash
% docker run --rm -it registry.suse.com/bci/golang:1.17 env
GOLANG_VERSION=1.17
PATH=/go/bin:/usr/local/go/bin:/root/go/bin/:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
TERM=xterm
HOME=/root
```

Reply:
@dcermak if I omit this now it will not build.

Reply:
@macedogm yeah, apparently the new version is not yet up